packages/next/src/utils.ts

```diff
@@ -2390,36 +2421,74 @@ export async function getMiddlewareManifest(
     return;
   }
 
-  return fs.readJSON(middlewareManifestPath);
+  const manifest = (await fs.readJSON(
+    middlewareManifestPath
+  )) as MiddlewareManifest;
+  return manifest.version === 1
+    ? upgradeMiddlewareManifest(manifest)
```

Comment:
**praise:** smart! I was about to ask how `getRouteMatchers` could be backward compatible. It does not have to be! 👏🏻